akka-cluster-typed/src/main/scala/akka/cluster/typed/ClusterSingleton.scala

```diff
@@ -61,37 +71,43 @@ final class ClusterSingletonSettings(
   def withNoDataCenter(): ClusterSingletonSettings = copy(dataCenter = None)
 
   def withRemovalMargin(removalMargin: FiniteDuration): ClusterSingletonSettings = copy(removalMargin = removalMargin)
+
   def withRemovalMargin(removalMargin: java.time.Duration): ClusterSingletonSettings =
     withRemovalMargin(removalMargin.asScala)
 
   def withHandoverRetryInterval(handOverRetryInterval: FiniteDuration): ClusterSingletonSettings =
     copy(handOverRetryInterval = handOverRetryInterval)
+
   def withHandoverRetryInterval(handOverRetryInterval: java.time.Duration): ClusterSingletonSettings =
     withHandoverRetryInterval(handOverRetryInterval.asScala)
 
   def withBufferSize(bufferSize: Int): ClusterSingletonSettings = copy(bufferSize = bufferSize)
 
+  def withLeaseSettings(leaseSettings: Option[LeaseUsageSettings]) = copy(leaseSettings = leaseSettings)
```

Comment:
The classic API is `withLeaseSettings(LeaseUsageSettings)` which makes it conveniently usable from both Java and Scala, and there never was any complaint about a way to set it to no lease after it already is lease so I think we can go with the same signature here. The field would still be an `Option`, just make this method wrap with `Some`.

Reply:
If you think being able to disable a pre-set lease is important, we need an additional method for Java users taking an `Optional[LeaseUsageSettings]`.

Reply:
Unsetting the lease programmatically might be an extreme edge case. 

That would only be useful, IMO, if you get a pre-configured cluster settings and you need to tweak it. I cannot imagine such scenario, except if you are using some kind of framework build on top of akka cluster, but in that case, you are probably no dealing with those details either. 

And, even if that's the case, you could do a manual clone of the cluster settings you get from said framework and not copy the lease settings. 

That said, I think we should go for: 

```scala
def withLeaseSettings(leaseSettings: LeaseUsageSettings) = 
  copy(leaseSettings = Option(leaseSettings)) 
```

btw, better `Option(leaseSettings)` then `Some(leaseSettings)`. If `leaseSettings` is null it becomes a `None`, instead of `Some(null)`.

Reply:
Reasonable, sorry that I didn't consider the java API and previous API. fixed.